### ISSUE TITLE
Check ObjectDeletedError catching on delete

### ISSUE THREAD

**Issue by Glutexo** [RedHatInsights/insights-host-inventory#400]
This looks like a possible bug, but maybe it isn’t. It’d be good to verify and comment.

When [deleting](https://github.com/RedHatInsights/insights-host-inventory/blob/aa5481588cc76459a08f2de78946c0ec125f9aea/api/host.py#L287) a Host, it can happen, that a host has been already deleted by a different request before [iteration](https://github.com/RedHatInsights/insights-host-inventory/blob/aa5481588cc76459a08f2de78946c0ec125f9aea/api/host.py#L287) finishes. I am not sure, whether the _sqlalchemy.orm.exc.ObjectDeletedError_ is actually raised when the _id_ property is [accessed](https://github.com/RedHatInsights/insights-host-inventory/blob/aa5481588cc76459a08f2de78946c0ec125f9aea/api/host.py#L289) or when [iterating](https://github.com/RedHatInsights/insights-host-inventory/blob/aa5481588cc76459a08f2de78946c0ec125f9aea/api/host.py#L287). The latter looks more probable to me, but I don’t know the internal workings of SQLAlchemy.

It’s also possible that https://github.com/RedHatInsights/insights-host-inventory/pull/392 will cause that the exception is never raised, because the property is no longer going to be accessed. Also I am not sure, what the _synchronize_session="fetch"_ argument for the [delete](https://github.com/RedHatInsights/insights-host-inventory/blob/aa5481588cc76459a08f2de78946c0ec125f9aea/api/host.py#L300) does and why there is an explicit [DB commit](https://github.com/RedHatInsights/insights-host-inventory/blob/aa5481588cc76459a08f2de78946c0ec125f9aea/api/host.py#L301).

All this should be properly commented and for that it’s necessary to research and verify the actual workings. Such a complex functionality also has to be covered by tests, although here multiprocessing is required and such test is going to be non-trivial.

**Comment by aleccohan** [comment]
It seems as though I was mistaken about my previous comment on this issue.  Through alot more testing I have found that without the changes made in PR #392 the _ObjectDeletedError_ is not catching at all.  In fact it continues to run even though the host has been deleted as if nothing happened at all.

**Comment by Glutexo** [comment]
Oh, so my original fear was correct. That‘s a bit sad, because it’s necessary to fix that so the error is correctly caught. But I’m glad you found it!

**Comment by aleccohan** [comment]
Yeah me too!  I was wondering why the test I was writing wasn't working so I decided to give it another look and it turns out your original idea was absolutely right.

I'm curious why the catch works with #392, I think you were absolutely right about that one as well with the accessing of the host during the printing of the deletion event.
